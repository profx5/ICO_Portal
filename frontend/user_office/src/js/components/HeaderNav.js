import React from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';



const HeaderNav = ({}) => (
    <Wrapper>
        <NavList>
            <NavItem className="active"><Link to="/user_office">Investors cabinet</Link></NavItem>
{/*            <NavItem><a>Referral & Bonuses</a></NavItem>
            <NavItem><a>Fundraiser's cabinet</a></NavItem>*/}
        </NavList>
    </Wrapper>
)

export default HeaderNav;

const Wrapper = styled.div`
    height: 100%;
`;

const NavList = styled.ul`
    height: 100%;
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
`;

const NavItem = styled.li`
    height: 100%;
    color: #222121;
    margin-left: 38px;
    white-space: nowrap;
    display: inline-flex;
    align-items: center;
    position: relative;
    &.active {
        color: #3172fd;
        &:before {
            content: '';
            position: absolute;
            bottom: 0;
            left: 0;
            height: 3px;
            width: 100%;
            background: #3172fd;
        }
    }
    a {
        display: flex;
        align-items: center;
        height: 100%;
    }
`;