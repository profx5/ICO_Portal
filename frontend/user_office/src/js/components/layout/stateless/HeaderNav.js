import React from 'react';
import styled from 'styled-components';
import {NavLink} from 'react-router-dom';
import {media} from 'js/utils/media';


const navigationItems = [
  {
    location: '/user_office/',
    caption: 'Investors cabinet',
    isActive: (match, location) => {
      if (!match) return false;
      return !location.pathname.startsWith('/user_office/referrals');
    }
  },
];

const HeaderNav = ({className}) => (
  <Wrapper className={className}>
    <NavList>
      {navigationItems.map(navigationItem => (
        <NavItem
            key={navigationItem.location}
            to={navigationItem.location}
            activeClassName='active' isActive={navigationItem.isActive}>
           {navigationItem.caption}
        </NavItem>
      ))}
    </NavList>
  </Wrapper>
);


export default HeaderNav;

const Wrapper = styled.div`
    height: 100%;
    ${media.xs} {
      background: #F5F6FA;
      padding-top: 15px;
      padding-bottom: 6px;
      height: 52px;
      overflow-x: scroll;
    }
`;

const NavList = styled.div`
    height: 100%;
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
`;

const NavItem = styled(NavLink)`
    height: 100%;
    font-weight: 400;
    color: #222121;
    margin-left: 38px;
    white-space: nowrap;
    display: inline-flex;
    align-items: center;
    position: relative;
    ${media.xs} {
      font-size: 14px;
      margin-left: 16px;
    }
    &.active {
        color: #3172fd;
        &:before {
            content: '';
            position: absolute;
            bottom: 0;
            left: 0;
            height: 3px;
            width: 100%;
            background: #3172fd;
        }
    }
`;
