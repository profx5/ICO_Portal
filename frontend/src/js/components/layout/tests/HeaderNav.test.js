import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import HeaderNav from 'js/components/layout/stateless/HeaderNav';


configure({ adapter: new Adapter() });

it('should render stateless HeaderNav', () => {
    const component = shallow(<HeaderNav/>);
    expect(component).toMatchSnapshot();
})
