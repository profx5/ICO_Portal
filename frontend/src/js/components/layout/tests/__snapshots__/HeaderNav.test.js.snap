// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless HeaderNav 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef
          activeClassName="active"
          isActive={[Function]}
          to="/user_office/"
        >
          Investors cabinet
        </ForwardRef>
      </ForwardRef>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            activeClassName="active"
            isActive={[Function]}
            to="/user_office/"
          >
            Investors cabinet
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "/user_office/",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "children": "Investors cabinet",
            "isActive": [Function],
            "to": "/user_office/",
          },
          "ref": null,
          "rendered": "Investors cabinet",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    height: 100%;
    font-weight: 400;
    color: #222121;
    margin-left: 38px;
    white-space: nowrap;
    display: inline-flex;
    align-items: center;
    position: relative;
    ",
                "@media (max-width: 767px)",
                " {
      font-size: 14px;
      margin-left: 16px;
    }
    &.active {
        color: #3172fd;
        &:before {
            content: '';
            position: absolute;
            bottom: 0;
            left: 0;
            height: 3px;
            width: 100%;
            background: #3172fd;
        }
    }
",
              ],
            },
            "displayName": "Styled(NavLink)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
    height: 100%;
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    height: 100%;
    ",
          "@media (max-width: 767px)",
          " {
      background: #F5F6FA;
      padding-top: 15px;
      padding-bottom: 6px;
      height: 52px;
      overflow-x: scroll;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef
            activeClassName="active"
            isActive={[Function]}
            to="/user_office/"
          >
            Investors cabinet
          </ForwardRef>
        </ForwardRef>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              activeClassName="active"
              isActive={[Function]}
              to="/user_office/"
            >
              Investors cabinet
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "/user_office/",
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "children": "Investors cabinet",
              "isActive": [Function],
              "to": "/user_office/",
            },
            "ref": null,
            "rendered": "Investors cabinet",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
    height: 100%;
    font-weight: 400;
    color: #222121;
    margin-left: 38px;
    white-space: nowrap;
    display: inline-flex;
    align-items: center;
    position: relative;
    ",
                  "@media (max-width: 767px)",
                  " {
      font-size: 14px;
      margin-left: 16px;
    }
    &.active {
        color: #3172fd;
        &:before {
            content: '';
            position: absolute;
            bottom: 0;
            left: 0;
            height: 3px;
            width: 100%;
            background: #3172fd;
        }
    }
",
                ],
              },
              "displayName": "Styled(NavLink)",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    height: 100%;
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    height: 100%;
    ",
            "@media (max-width: 767px)",
            " {
      background: #F5F6FA;
      padding-top: 15px;
      padding-bottom: 6px;
      height: 52px;
      overflow-x: scroll;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
